packages/block-library/src/more/edit.native.js

```diff
@@ -11,40 +11,29 @@ import { HorizontalRule } from '@wordpress/components';
  */
 import styles from './editor.scss';
 
-export class MoreEdit extends Component {
-	constructor() {
-		super( ...arguments );
+function MoreEdit( { attributes, getStylesFromColorScheme } ) {
+	const [ defaultText ] = useState( __( 'Read more' ) );
+	const { customText } = attributes;
 
-		this.state = {
-			defaultText: __( 'Read more' ),
-		};
-	}
+	const content = customText || defaultText;
+	const textStyle = getStylesFromColorScheme(
+		styles.moreText,
+		styles.moreTextDark
+	);
+	const lineStyle = getStylesFromColorScheme(
+		styles.moreLine,
+		styles.moreLineDark
+	);
 
-	render() {
-		const { attributes, getStylesFromColorScheme } = this.props;
-		const { customText } = attributes;
-		const { defaultText } = this.state;
-
-		const content = customText || defaultText;
-		const textStyle = getStylesFromColorScheme(
-			styles.moreText,
-			styles.moreTextDark
-		);
-		const lineStyle = getStylesFromColorScheme(
-			styles.moreLine,
-			styles.moreLineDark
-		);
-
-		return (
-			<HorizontalRule
-				text={ content }
-				marginLeft={ 0 }
-				marginRight={ 0 }
-				textStyle={ textStyle }
-				lineStyle={ lineStyle }
-			/>
-		);
-	}
+	return (
+		<HorizontalRule
+			text={ content }
```

Comment:
```suggestion
			text={ customText ||  __( 'Read more' ) }
```
There's no need for the state to hold a static fallback value. We can just inline this.